src/bench/wallet_migration.cpp

```diff
@@ -68,6 +68,9 @@ static void WalletMigration(benchmark::Bench& bench)
         assert(res);
         assert(res->wallet);
         assert(res->watchonly_wallet);
+
+        res->wallet->Close();
+        res->watchonly_wallet->Close();
```

Comment:
Should the fact that the migrated wallet databases remain open be documented in `MigrateLegacyToDescriptor`?

Reply:
I needed to close both for the build to pass (`res->wallet->Close()` alone wasn't enough).
Maybe it passes with only `res->watchonly_wallet->Close()` - I can check if you think it makes sense.

Reply:
I wonder why `res->watchonly_wallet.use_count() == 3`? Is this the expected behavior of the `MigrateLegacyToDescriptor()` function?

Reply:
Maybe @furszy knows.

Reply:
Unfortunately, this patch does not fix the test case posted [here](https://github.com/bitcoin/bitcoin/issues/32291#issuecomment-2816658407).

Reply:
> Maybe it passes with only res->watchonly_wallet->Close() - I can check, if you think it makes sense

No, we do need to close both: https://github.com/l0rinc/bitcoin/actions/runs/14548535133/job/40817048030?pr=9

Reply:
> Unfortunately, this patch does not fix the test case posted https://github.com/bitcoin/bitcoin/issues/32291#issuecomment-2816658407.

I can reproduce the failure with
```bash
cmake --preset=libfuzzer -DBUILD_FOR_FUZZING=OFF -DBUILD_BENCH=ON && cmake --build build_fuzz -j$(nproc) && build_fuzz/bin/bench_bitcoin -filter='WalletMigration' --min-time=10000
```

I pushed a new version based on https://github.com/bitcoin/bitcoin/blob/master/src/bench/wallet_create.cpp#L54 which calls RemoveWallet instead - @hebasto, can you please check if this fixes it for you as well?

Reply:
Pushed again, please recheck